magefiles/dagger/precommit/dagger.go

```diff
@@ -0,0 +1,48 @@
+package precommit
+
+import (
+	_ "embed"
+
+	"context"
+
+	"dagger.io/dagger"
+)
+
+//go:embed scripts/install_dependencies.sh
+var InstallDependenciesScript string
+
+func Run(ctx context.Context, client *dagger.Client, workdir *dagger.Directory) error {
```

Comment:
Loving it! one question - how do we test these?

Reply:
Sorry for late answer, we're still discovering test process.